rotkehlchen/chain/ethereum/modules/nfts.py

```diff
@@ -312,6 +319,19 @@ def delete_price_for_nft(self, asset: Asset) -> bool:
 
         return True
 
+    def _filter_ignored_nfts(self, nfts_data: Dict[ChecksumEvmAddress, List[NFT]]) -> Dict[ChecksumEvmAddress, List[NFT]]:  # noqa: E501
+        """Remove ignored NFTs from NFTs data."""
+        with self.db.conn.read_ctx() as cursor:
+            # convert to set to allow O(1) during `in` conditional below.
+            ignored_nfts = set(self.db.get_ignored_assets(cursor=cursor, only_nfts=True))
+
+        # make a copy to avoid mutating the cached data.
+        nfts_data_copy = nfts_data.copy()
```

Comment:
What cache? This is the passed argument.

Reply:
Copying from discord: 

The cache mention here confuses me. Are you trying to tell me that any function with cache_response_timewise is potentially used wrong in the codebase? In the sense that any function that has cache_response_timewise returns a mutable response which is potentially mutated by any caller? 
If that is true then that modifier is terrible and will bite us in the ass and if at any place a copy is made it should be inside the decorator itself? Would need to think about it.

Reply:
> Copying from discord:
> 
> The cache mention here confuses me. Are you trying to tell me that any function with cache_response_timewise is potentially used wrong in the codebase? In the sense that any function that has cache_response_timewise returns a mutable response which is potentially mutated by any caller? If that is true then that modifier is terrible and will bite us in the ass and if at any place a copy is made it should be inside the decorator itself? Would need to think about it.

Yeah, pretty much. 
I suppose the reason why we've not had an issue is that... we don't manipulate the data, it's just returned -- read only.

Reply:
I would say the right approach here is to do what I said. Perhaps I can just do it in another PR and merge this one. Or however you prefer. Will need to think about it.

Reply:
That works I guess